### PR TITLE
support choosing non-pending tiflash peer when buildBatchCopTask

### PR DESCRIPTION
**Pull request by guo-shaoge**
issue: https://github.com/pingcap/tiflash/issues/7301

Before: Task can dispatch to store whose region is still pending.
After: Try not to dispatch task to pending store.

What's changed:
1. Change function declaration of `getRegionByKey()` and `getRegionByID()`, return whole response instead of only region info.
2. Record pending store of region in region cache.
3. Try dispatch task to non_pending_store of region when buildBatchCopTasks.

### CODE REVIEW

File: include/pingcap/kv/RegionCache.h [hunk 1]
```diff
@@ -84,6 +85,8 @@ struct Region
 {
     metapb::Region meta;
     metapb::Peer leader_peer;
+    // This is just an indicator, maybe empty.
+    std::vector<metapb::Peer> pending_peers;
```
Comment (guo-shaoge) [inline include/pingcap/kv/RegionCache.h:89]: check onStaleRead

File: include/pingcap/kv/RegionCache.h [hunk 2]
```diff
@@ -195,7 +204,7 @@ class RegionCache
 
     Store getStore(Backoffer & bo, uint64_t id);
 
-    std::vector<uint64_t> getAllValidTiFlashStores(Backoffer & bo, const RegionVerID & region_id, const Store & current_store, const LabelFilter & label_filter);
+    std::pair<std::vector<uint64_t>, std::vector<uint64_t>> getAllValidTiFlashStores(Backoffer & bo, const RegionVerID & region_id, const Store & current_store, const LabelFilter & label_filter);
```
Comment (JaySon-Huang) [inline include/pingcap/kv/RegionCache.h:207]: Add some comments about the meaning of the return type.

**Review by JaySon-Huang — APPROVED** [review]
Rest LGTM

File: include/pingcap/kv/RegionCache.h (hunk 2 above)
Comment (guo-shaoge) [reply include/pingcap/kv/RegionCache.h:210]: Done

**Review by gengliqi — APPROVED** [review]

**Pull request merged**